PIP install build error

I keep getting this error while trying to install to WSL. I have numpy and other requirements. 

`     note: This error originates from a subprocess, and is likely not a problem with pip.
        ERROR: Failed building wheel for numpy
      Failed to build numpy
      ERROR: Could not build wheels for numpy, which is required to install pyproject.toml-based projects
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.`

I had the same problem:
"C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.32.31326\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -DHAVE_CONFIG_H -DDIC_VERSION=102 -DMECAB_DEFAULT_RC=\"dummy\" -DPACKAGE=\"open_jtalk\" -DVERSION=\"1.10\" -DCHARSET_UTF_8 -IC:\Users\ADMINI~1\AppData\Local\Temp\pip-build-env-d30lyddm\overlay\Lib\site-packages\numpy\core\include -IH:\secrectFG\pyopenjtalk\lib\open_jtalk\src\jpcommon -IH:\secrectFG\pyopenjtalk\lib\open_jtalk\src\mecab/src -IH:\secrectFG\pyopenjtalk\lib\open_jtalk\src\mecab2njd -IH:\secrectFG\pyopenjtalk\lib\open_jtalk\src\njd -IH:\secrectFG\pyopenjtalk\lib\open_jtalk\src\njd2jpcommon -IH:\secrectFG\pyopenjtalk\lib\open_jtalk\src\njd_set_accent_phrase -IH:\secrectFG\pyopenjtalk\lib\open_jtalk\src\njd_set_accent_type -IH:\secrectFG\pyopenjtalk\lib\open_jtalk\src\njd_set_digit -IH:\secrectFG\pyopenjtalk\lib\open_jtalk\src\njd_set_long_vowel -IH:\secrectFG\pyopenjtalk\lib\open_jtalk\src\njd_set_pronunciation -IH:\secrectFG\pyopenjtalk\lib\open_jtalk\src\njd_set_unvoiced_vowel -IH:\secrectFG\pyopenjtalk\lib\open_jtalk\src\text2mecab -IC:\Users\Administrator\.conda\envs\py37tf\include -IC:\Users\Administrator\.conda\envs\py37tf\include "-IC:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.32.31326\ATLMFC\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.32.31326\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\cppwinrt" /EHsc /Tplib\open_jtalk\src\mecab/src\utils.cpp /Fobuild\temp.win-amd64-3.7\Release\lib\open_jtalk\src\mecab/src\utils.obj /source-charset:utf-8 /execution-charset:utf-8
      utils.cpp
      lib\open_jtalk\src\mecab/src\utils.cpp(60): fatal error C1083: 无法打开包括文件: “dirent.h”: No such file or directory
      setup.py:26: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
        _CYTHON_INSTALLED = ver >= LooseVersion(min_cython_ver)
      error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Professional\\VC\\Tools\\MSVC\\14.32.31326\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pyopenjtalk
Failed to build pyopenjtalk
ERROR: Could not build wheels for pyopenjtalk, which is required to install pyproject.toml-based projects



My environment:

Windows 10 (x64)
Visual Studio 2022

i got the same issue, is your issue fixed? if so, would share the practice? thx

> I had the same problem: "C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.32.31326\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -DHAVE_CONFIG_H -DDIC_VERSION=102 -DMECAB_DEFAULT_RC="dummy" -DPACKAGE="open_jtalk" -DVERSION="1.10" -DCHARSET_UTF_8 -IC:\Users\ADMINI~1\AppData\Local\Temp\pip-build-env-d30lyddm\overlay\Lib\site-packages\numpy\core\include -IH:\pyopenjtalk\lib\open_jtalk\src\jpcommon -IH:\pyopenjtalk\lib\open_jtalk\src\mecab/src -IH:\pyopenjtalk\lib\open_jtalk\src\mecab2njd -IH:\pyopenjtalk\lib\open_jtalk\src\njd -IH:\pyopenjtalk\lib\open_jtalk\src\njd2jpcommon -IH:\pyopenjtalk\lib\open_jtalk\src\njd_set_accent_phrase -IH:\pyopenjtalk\lib\open_jtalk\src\njd_set_accent_type -IH:\pyopenjtalk\lib\open_jtalk\src\njd_set_digit -IH:\pyopenjtalk\lib\open_jtalk\src\njd_set_long_vowel -IH:\pyopenjtalk\lib\open_jtalk\src\njd_set_pronunciation -IH:\pyopenjtalk\lib\open_jtalk\src\njd_set_unvoiced_vowel -IH:\pyopenjtalk\lib\open_jtalk\src\text2mecab -IC:\Users\Administrator.conda\envs\py37tf\include -IC:\Users\Administrator.conda\envs\py37tf\include "-IC:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.32.31326\ATLMFC\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.32.31326\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt" /EHsc /Tplib\open_jtalk\src\mecab/src\utils.cpp /Fobuild\temp.win-amd64-3.7\Release\lib\open_jtalk\src\mecab/src\utils.obj /source-charset:utf-8 /execution-charset:utf-8 utils.cpp lib\open_jtalk\src\mecab/src\utils.cpp(60): fatal error C1083: 无法打开包括文件: “dirent.h”: No such file or directory setup.py:26: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead. _CYTHON_INSTALLED = ver >= LooseVersion(min_cython_ver) error: command 'C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.32.31326\bin\HostX86\x64\cl.exe' failed with exit code 2 [end of output]
> 
> note: This error originates from a subprocess, and is likely not a problem with pip. ERROR: Failed building wheel for pyopenjtalk Failed to build pyopenjtalk ERROR: Could not build wheels for pyopenjtalk, which is required to install pyproject.toml-based projects
> 
> My environment:
> 
> Windows 10 (x64) Visual Studio 2022

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -2,6 +2,7 @@
 requires = [
     "setuptools<v60.0",
     "cython>=0.28.0,<3.0.0",
+    "cmake",
     "numpy>=1.25.0; python_version>='3.9'",
     "oldest-supported-numpy; python_version<'3.9'",
 ]

